community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/cleanup_compute/cleanup_compute.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set -e -o pipefail
+
+project="$1"
+cluster_name="$2"
+nodeset_name="$3"
+universe_domain="$4"
+compute_endpoint_version="$5"
+gcloud_dir="$6"
+
+if [[ -z "${project}" || -z "${cluster_name}" || -z "${nodeset_name}" || -z "${universe_domain}" || -z "${compute_endpoint_version}" ]]; then
+	echo "Usage: $0 <project> <cluster_name> <nodeset_name> <universe_domain> <compute_endpoint_version> <gcloud_dir>"
+	exit 1
+fi
+
+if [[ -n "${gcloud_dir}" ]]; then
+	export PATH="$gcloud_dir:$PATH"
+fi
+
+export CLOUDSDK_API_ENDPOINT_OVERRIDES_COMPUTE="https://www.${universe_domain}/compute/${compute_endpoint_version}/"
+export CLOUDSDK_CORE_PROJECT="${project}"
+
+if ! type -P gcloud 1>/dev/null; then
+	echo "gcloud is not available and your compute resources are not being cleaned up"
+	echo "https://console.cloud.google.com/compute/instances?project=${project}"
+	exit 1
+fi
+
+echo "Deleting compute nodes"
+node_filter="name:${cluster_name}-${nodeset_name}-* labels.slurm_cluster_name=${cluster_name} AND labels.slurm_instance_role=compute"
+
+tmpfile=$(mktemp)
+running_nodes_filter="${node_filter} AND (status!=STOPPING AND status!=TERMINATED)"
+# List all currently running instances and attempt to delete them
+gcloud compute instances list --format="value(selfLink)" --filter="${running_nodes_filter}" >"$tmpfile"
+# Do 10 instances at a time
+while batch="$(head -n 10)" && [[ ${#batch} -gt 0 ]]; do
+	nodes=$(echo "$batch" | paste -sd " " -) # concat into a single space-separated line
+	# The lack of quotes around ${nodes} is intentional and causes each new space-separated "word" to
+	# be treated as independent arguments. See PR#2523
+	# shellcheck disable=SC2086
+	gcloud compute instances delete --quiet ${nodes} || echo "Failed to delete some instances"
+done <"$tmpfile" # have to use a temp file, since `< <(gcloud ...)` doesn't work nicely with `head`
+rm -f "$tmpfile"
```

Comment:
I suggest a `trap` for this immediately after creating the file. e.g.

`trap 'rm -rf -- "$tmpfile"' EXIT`

(confirm this first, I'm doing a bit of cut-and-paste from example)